extension/content.js

```diff
@@ -108,16 +108,39 @@ function getVideoId() {
     }
 }
 
-function fetchComments(videoId) {
-    const part = 'snippet'
-    const fields = 'items(snippet(topLevelComment(snippet)))'
-    const order = 'relevance'
-    return fetch(`https://www.googleapis.com/youtube/v3/commentThreads?videoId=${videoId}&part=${part}&fields=${fields}&order=${order}&key=${API_KEY}`)
+function fetchAllComments(videoId) {
+    return new Promise(async (resolve) => {
+        var items = [];
+        await fetchComments(videoId, items).then((res) => {
+            return resolve(res);
+        })
+    })
+}
+
+function fetchComments(videoId, items, pageToken) {
+    return new Promise((resolve) => {
+        const part = 'snippet'
+        const fields = 'items(snippet(topLevelComment(snippet))),nextPageToken'
+        const order = 'relevance'
+        const maxResults = 100
+
+        var opts = '';
+        if (pageToken) {
+            opts = `&pageToken=${pageToken}`
+        }
+
+        fetch(`https://www.googleapis.com/youtube/v3/commentThreads?videoId=${videoId}&part=${part}&fields=${fields}&order=${order}&maxResults=${maxResults}&key=${API_KEY}` + opts)
```

Comment:
Could you move concatenation outside the parameters?
```
var url = ...
if (pageToken) {
  url += ...
}
```

Reply:
Fixed in last commit